pi-manage

```diff
@@ -949,28 +949,39 @@ def create(name, scope, action, filename=None):
         return r
 
 
-@api_manager.command
-def createtoken(role=ROLE.ADMIN):
+@api_manager.option('-r', '--role', help="The role of the API key can either be"
+                                         "'admin' or 'validate' to access the admin"
+                                         "API or the validate API.")
+@api_manager.option('-d', '--days', help='The number of days the access token should be valid.'
+                                         ' Defaults to 365.')
+@api_manager.option('-R', '--realm', help='The realm of the admin. Defaults to "API"')
+@api_manager.option('-u', '--username', help='The username of the admin.')
+def createtoken(role=ROLE.ADMIN, days=365, realm="API", username=None):
```

Comment:
Oh no, seems like manager.option loses the default arguments. We need to specify them in the .option decorators, because right now:
```
╰─$ ./pi-manage api createtoken -r admin                                                   1 ↵

             _                    _______  _______
   ___  ____(_)  _____ _______ __/  _/ _ \/ __/ _ |
  / _ \/ __/ / |/ / _ `/ __/ // // // // / _// __ |
 / .__/_/ /_/|___/\_,_/\__/\_, /___/____/___/_/ |_|
/_/                       /___/
   
Traceback (most recent call last):
  File "./pi-manage", line 996, in <module>
    manager.run()
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask_script/__init__.py", line 412, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask_script/__init__.py", line 383, in handle
    res = handle(*args, **config)
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "./pi-manage", line 971, in createtoken
    validity = timedelta(days=int(days))
TypeError: <flask_script.commands.Command object at 0x7f323b564350>: int() argument must be a string or a number, not 'NoneType'
```